components/studentPortal/freemium/FreemiumDialogComponent.tsx

```diff
@@ -0,0 +1,105 @@
+import { XIcon } from "@heroicons/react/outline";
+import { Close, Content, Overlay, Portal, Root } from "@radix-ui/react-dialog";
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { getModalContent } from "../../../pages/api/studentPortal/freemium/getModalContent";
+import { themeSelector } from "../../../redux/themeSlice";
+import TrailAnimation from "../../ui/TrailAnimation";
+
+export enum ModalStage {
+  ONE,
+  TWO,
+  THREE,
+  FOUR,
+  FIVE,
+}
+
+const FreemiumDialogComponent: React.FC = () => {
+  const [currentStage, setCurrentStage] = useState(ModalStage.ONE);
+  const { currentTheme } = useSelector(themeSelector);
+  const [activeModal, setActiveModal] = useState(ModalStage.ONE);
+
+  const activeModalStyling = (currentStage: ModalStage) => {
+    let styling = "md:w-6 md:h-6 w-4 h-4 rounded-full";
+    if (currentStage === activeModal) {
+      styling = styling + " bg-rattata";
+    } else {
+      styling = styling + " bg-gray-300";
+    }
+    return styling;
+  };
+
+  useEffect(() => {
+    const handleKeyPress = (event: KeyboardEvent) => {
```

Comment:
i think there's some opportunity to clean this function up, or at the least add some comments

Reply:
I added comments, am open to cleaning it up more as well.